services/secondary-storage/server/src/main/java/org/apache/cloudstack/storage/template/DownloadManagerImpl.java

```diff
@@ -304,8 +317,27 @@ public void setDownloadStatus(String jobId, Status status) {
                     td.setStatus(Status.POST_DOWNLOAD_FINISHED);
                     td.setDownloadError("Install completed successfully at " + new SimpleDateFormat().format(new Date()));
                 }
-            }
-            else {
+            } else if (td instanceof SwiftVolumeDownloader) {
+                dj.setCheckSum(((SwiftVolumeDownloader) td).getMd5sum());
+                if ("vhd".equals(((SwiftVolumeDownloader) td).getFileExtension()) ||
```

Comment:
```suggestion
                if ("vhd".equalsIgnoreCase(((SwiftVolumeDownloader) td).getFileExtension()))
```

Ignore if it is strictly all lower/upper case.